artifact-post-action/compute-risk-and-residual/src/main.rs

```diff
@@ -0,0 +1,205 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2024 Enalean
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+use serde::{Serialize, Deserialize};
+use serde_json::json;
+use std::error::Error;
+use std::io::stdin;
+use std::str::FromStr;
+
+#[derive(Serialize, Debug)]
+struct FieldValueBinding {
+    field_id: i64,
+    bind_value_ids: Vec<i64>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ListValue {
+    id: Option<i64>,
+    label: Option<String>,
+    color: Option<String>,
+    tlp_color: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "type")]
+enum FieldValue {
+    #[serde(rename = "string")]
+    String {
+        field_id: u32,
+    },
+    #[serde(rename = "text")]
+    Text {
+        field_id: u32,
+    },
+    #[serde(rename = "art_link")]
+    Link {
+        field_id: u32,
+    },
+    #[serde(rename = "sb")]
+    SelectBox {
+        field_id: u32,
+        label: String,
+        values: Vec<ListValue>,
+        bind_value_ids: Vec<u32>,
+    },
+    #[serde(rename = "aid")]
+    ArtifactId {
+        field_id: u32,
+        label: String,
+    },
+    #[serde(rename = "rb")]
+    RadioButton {
+        field_id: u32,
+        label: String,
+    },
+}
+
+#[derive(Serialize, Deserialize)]
+struct Changeset {
+    id: i64,
+    values: Vec<FieldValue>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct TrackerFieldValue {
+    id: i64,
+    label: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct TrackerField {
+    field_id: i64,
+    label: String,
+    values: Option<Vec<TrackerFieldValue>>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct Tracker {
+    id: i64,
+    fields: Vec<TrackerField>
+}
+
+#[derive(Serialize, Deserialize)]
+struct Artifact {
+    id: i64,
```

Comment:
`i64` seems a bit overkill. `u32` should be enough as mysql int default size is 32 and we don't have negative integer